### PR TITLE
chore(e2e): improve monitor only logic

### PR DESCRIPTION
**Pull request by corverroos**
Refactor monitor-only logic to have less inline branching. Prefer splitting logic: ``if OnlyMonitor, do something else and return"

task: none

### CODE REVIEW

File: e2e/app/definition.go [hunk 1]
```diff
@@ -322,45 +322,6 @@ func publicChains(manifest types.Manifest, cfg DefinitionConfig) ([]types.Public
 func internalNetwork(testnet types.Testnet, deployInfo map[types.EVMChain]netman.DeployInfo, evmPrefix string) netconf.Network {
 	var chains []netconf.Chain
 
-	// in monitor only mode there are no anvil chains and omni EVMs
-	if !testnet.OnlyMonitor {
```
Comment (corverroos) [inline e2e/app/definition.go:326]: move this to the bottom

File: e2e/app/run.go [hunk 2]
```diff
@@ -39,69 +39,62 @@ func Deploy(ctx context.Context, def Definition, cfg DeployConfig) (types.Deploy
 		return nil, nil, err
 	}
 
-	genesisValSetID := uint64(1) // validator set IDs start at 1
-	var genesisVals []bindings.Validator
-	var err error
-	if !def.Manifest.OnnyMonitor {
```
Comment (corverroos) [inline e2e/app/run.go:45]: Only a single `if def.Manifest.OnnyMonitor` at the top

File: e2e/app/definition.go [hunk 3]
```diff
@@ -378,6 +339,50 @@ func internalNetwork(testnet types.Testnet, deployInfo map[types.EVMChain]netman
 		chains = append(chains, pc)
 	}
 
+	// In monitor only mode, there is only public chains, so skip omni and anvil chains.
+	if testnet.OnlyMonitor {
+		return netconf.Network{
+			Name:   testnet.Network,
+			Chains: chains,
+		}
+	}
```
Comment (Lazar955) [inline e2e/app/definition.go:348]: 😍

**Review by Lazar955 — APPROVED** [review]
nice cleanup

File: e2e/app/run.go [hunk 4]
```diff
@@ -243,3 +236,17 @@ func logRPCs(ctx context.Context, def Definition) {
...
+	if err := Setup(ctx, def, cfg.AgentSecrets, cfg.testConfig); err != nil {
+		return err
+	}
+
+	if err := def.Infra.StartNodes(ctx); err != nil {
+		return errors.Wrap(err, "starting initial nodes")
+	}
+
+	return nil
+}
```
Comment (pavelnikolov) [inline e2e/app/run.go:252]: looks much better indeed, thanks

**Pull request merged**